controller/api/v1alpha1/managednamespace_types.go

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	console "github.com/pluralsh/console-client-go"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ManagedNamespaceSpec defines the desired state of ManagedNamespace
+type ManagedNamespaceSpec struct {
+	// Name of this namespace once its placed on a cluster. If not provided ManagedNamespace's own name from ManagedNamespace.ObjectMeta will be used.
+	// +kubebuilder:validation:Optional
+	Name *string `json:"name,omitempty"`
+	// Description a short description of the purpose of this namespace
+	// +kubebuilder:validation:Optional
+	Description *string `json:"description,omitempty"`
+	// Labels for this namespace
+	// +kubebuilder:validation:Optional
+	Labels map[string]string `json:"labels,omitempty"`
+	// Annotations for this namespace
+	// +kubebuilder:validation:Optional
+	Annotations map[string]string `json:"annotations,omitempty"`
+	// PullSecrets a list of pull secrets to attach to this namespace
+	// +kubebuilder:validation:Optional
+	PullSecrets []string `json:"pullSecrets,omitempty"`
+	// +kubebuilder:validation:Optional
+	Service *ServiceTemplateAttributes `json:"service,omitempty"`
+	// +kubebuilder:validation:Optional
+	Target *ClusterTargetAttributes `json:"target,omitempty"`
+}
+
+// A spec for targeting clusters
+type ClusterTargetAttributes struct {
```

Comment:
I feel like the names for this and `ServiceTemplateAttributes` don't need the attributes suffix (we only use that in gql because input and regular objects need distinct type names)

Reply:
done